paasta_tools/cli/utils.py

```diff
@@ -731,6 +731,27 @@ def validate_given_deploy_groups(all_deploy_groups, args_deploy_groups):
     return valid_deploy_groups, invalid_deploy_groups
 
 
+def short_to_full_git_sha(short, refs):
+    """Converts a short git sha to a full sha
+
+    :param short: A short git sha represented as a string
+    :param refs: A list of refs in the git repository
+    :return: The full git sha or None if one can't be found
+    """
+    for sha in list(set(refs.values())):
+        if sha.startswith(short):
+            return sha
+    return None
```

Comment:
Unless we make this another interactive prompt (which I'm still not a huge fan of), returning a list seems like more trouble than it is worth. Or I guess another option would be to go with the list and make the caller raise an exception if the list doesn't have exactly 1 element.

The list(set( stuff was leftover from some earlier testing. I'll get rid of it.